### PR TITLE
display more useful output on errors #187

### LINKED ISSUE #187: display more useful output on errors
In case of an error somewhere in the `helix-pipeline`, the browser output is a bleak `""`. The developer will then have to look at the console output/log to find out what went wrong. 

Let's try to create a more meaningful response in case of an error.

(Partially implements https://github.com/adobe/helix-simulator/issues/17)

### PR DESCRIPTION
**Pull request by rofe**
Fixes #187

### CODE REVIEW

**Comment by trieloff** [comment]
See also https://github.com/adobe/helix-simulator/issues/17

**Comment by rofe** [comment]
> Please check that the error output is only shown in non-production environments

Like so?

File: src/html/set-status.js [hunk 1]
```diff
@@ -19,12 +20,16 @@ function setStatus({ response = {}, error }, { logger }) {
...
+    const isDev = !production();
+    const res = {};
+    res.status = 500;
+    if (isDev) {
+      res.body = `<html><body><h1>500</h1><p>${error}</p></body></html>`;
+      res.headers = { 'Content-Type': 'text/html' };
+    } else {
+      res.body = '';
+    }
+    return res;
```
Comment (tripodsan) [inline src/html/set-status.js]: you're missing a `response` object in your response :-)

```suggestion
    return {
       response: res,
    };
```

File: test/testSetStatus.js [hunk 2]
```diff
@@ -20,16 +20,33 @@ const logger = Logger.getTestLogger({
...
+
   it('sets a 500 for an error', () => {
     assert.deepEqual(
-      setStatus({ content: { html: '<html></html>' }, error: 'oh, no!' }, { logger }),
+      setStatus({ content: { html: '<html></html>' }, error }, { logger }),
       {
-        response: {
-          status: 500,
-          body: '',
+        status: 500,
```
Comment (tripodsan) [inline test/testSetStatus.js]: is this really correct? don't we need the `response` object?

File: src/html/set-status.js (hunk 1 above)
Comment (rofe) [reply src/html/set-status.js]: Oops :)

File: test/testSetStatus.js (hunk 2 above)
Comment (rofe) [reply test/testSetStatus.js]: you're right.

**Comment by rofe** [comment]
Readded lost `response` object (thanks @tripodsan!)

**Comment by trieloff** [comment]
@rofe it's generally a better idea to pull the `if` statement out of the function and apply it in the pipeline like this:

```javascript
.error(detailed-status).when(() => !production())
.error(terse-status).when(() => production())
```

that way, your `detailed-status` and `terse-status` functions can stay simple, and you easily get 100% branch coverage in testing because there is only one branch in each function.

**Pull request merged**